### PR TITLE
Upgraded content-api-client to use retry-backoff feature.

### PR DESCRIPTION
**Pull request by anshulbajpai**

### CODE REVIEW

File: app/com/gu/itunes/CustomCapiClient.scala [hunk 1]
```diff
@@ -33,6 +36,9 @@ class CustomCapiClient(val apiKey: String) extends ContentApiClient {
     }
   }
 
+  override implicit val executor: ScheduledExecutor = ScheduledExecutor()
+  private val initialDelay = 250.millis
+  override val backoffStrategy: ContentApiBackoff = ContentApiBackoff.constantStrategy(initialDelay, maxAttempts = 3)
```
Comment (anshulbajpai) [inline app/com/gu/itunes/CustomCapiClient.scala]: Keeping it same as dotcom

File: build.sbt [hunk 2]
```diff
@@ -5,18 +5,16 @@ description   := "podcasts RSS feed"
...
   .settings(
     libraryDependencies ++= Seq(
       "org.jsoup" % "jsoup" % "1.8.1",
-      "com.gu" %% "content-api-client" % capiVersion,
+      "com.gu" %% "content-api-client" % "15.7",
       "com.squareup.okhttp3" % "okhttp" % "3.11.0",
       "org.scalactic" %% "scalactic" % "3.0.5",
       "org.scalatest" %% "scalatest" % "3.0.5" % "test",
-      "com.gu" %% "content-api-models-json" % capiVersion % "test"
+      "com.gu" %% "content-api-models-json" % "15.5" % "test"
```
Comment (anshulbajpai) [inline build.sbt]: I had to upgrade the capi models json version as it wasn't compatible with latest capi client version. Is it safe to do so?

File: app/com/gu/itunes/CustomCapiClient.scala (hunk 1 above)
Comment (regiskuckaertz) [reply app/com/gu/itunes/CustomCapiClient.scala]: Same here, should be exponential. The constant strategy was probably a mistake, it is useless until we have a means to inspect the exact reason for the error.

File: build.sbt (hunk 2 above)
Comment (regiskuckaertz) [reply build.sbt]: Yes.

**Pull request merged**